### ISSUE TITLE
REQUEST: A simple way to add new picture (\fig) fields into PTXprint without wiping out existing settings 

### ISSUE THREAD

**Issue by davidc86** [sillsdev/ptx2pdf#787]
We just added 3 new pictures in a book in Paratext. But we already had set up all the other pictures in PTXprint for that book beforehand.
What we need is a way to ADD the new figures without messing with the existing settings in PTXprint. 
The current ADD button is basically a "DIY" approach, whereas the GENERATE "Append" option basically clears the entire list for that book (or worse if not paying attention) and RESETS ever picture from scratch.

I have no idea how tricky it would be to do this, likely not trivial, but it will become more and more important for everyone.

I figure that if the picture _filename_ is the same in PTXprint as what was found in Paratext, and if the "_Caption Reference_" is the same, then the import process just skips that picture entry, because apparently it is already accounted for in PTXprint.

 This would be a huge help. 
(The "update captions" feature has already been a big help! Used it again just last night. Thank you for providing that option.)

**Comment by JDNystrom** [comment]
I agree this would be a huge help.

**Comment by jamespostsil** [comment]
I also agree this would be helpful, or perhaps the ability for PTXprint to recognize when an image has been added and automatically "import" it into PTXprint? I don't know how that would affect other images already in PTXprint.

**Comment by davidg-sil** [comment]
I've no idea how easy or not this might be. What I'd like to suggest is that when there's a prior piclist and the user click 'Generate', PTXprint offers a two lists, say new at the top and old below. When the user selects multiple lines from the new, they can select  "Import line", "Use all new".
When the user selects one entry from the top and one from the bottom, then perhaps it would 
highlighting changed bits: figure names, references, etc.
Then the user might have the option to   "Replace entry", "Update caption from selected Paratext entry", "update image from Paratext entry" and similar options. Thus the process would still be manual but it would be much controlled than  'destroy my work and start again', or click on the add button and don't mistype the caption that we have at the moment.

**Comment by markpenny** [comment]
@davidc86  This feature has now been added in 2.5.6 - so that if you click on the **Generate...** button, there is an option (enabled by default) to preserve your existing settings. Something like this below:
![2024-08-08 16_45_01-PTXprint 2 5 5  -  WSGlatin__Ruth_____Cloze  (RUT)](https://github.com/user-attachments/assets/e40dc962-8702-4a2b-92e3-baea0ac58581)

Please try it out and let us know what (else) we might need to improve in this area. Also, is the Refresh List button now redundant? And if we were to have a single button (instead of **Generate...** and **Refresh List**), what should we call it?

**Comment by davidc86** [comment]
Cool! Thanks for putting that in.
I think I actually used this once for another project. Very helpful. (So #787 can be closed.)

David
________________________________

**Comment by davidc86** [comment]
It is spooky to accidentally wipe out a PIC List settings file, and it is not always obvious what "REFRESH" means – like which does what to which? Does the Paratext SFM files push a repopulation to the Pic Settings list, filling in only those bits that are different? Does it pitch the List and rebuild it based on the current Paratext SFM files? Does it read through the entire project's set of SFM files (OT and NT) or only the current selected books?

Yeah, lots of spooky bits there.

I might see an advantage of the ONE button approach, but as is, the pop-up help currently makes it pretty clear that one button just updates the captions and location references, and nothing else, whereas the other button can do serious changes to your Pic List Settings.

But if there is only one button and in that button's window is a set of options that are worded nicely enough (with helps) to make it clear what will happen with whatever one is selected then that might work to. As what to call it, maybe "Picture List" and in there would be "Generate New List" "Update List" "Append List" "Refresh Captions" etc. much like what you have now, but with the added options from the other button.

David
________________________________
